packages/docusaurus-theme-classic/src/theme/hooks/useDocs.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// See https://github.com/facebook/docusaurus/issues/3360
+// TODO find a better solution, this shouldn't be needed
+export {};
+/*
+throw new Error(
+  "The docs plugin is not used, so you can't require the useDocs hooks. ",
+);
+ */
```

Comment:
We seem to assume here that this dummy theme-classic file is the fallback, and the real hooks in plugin-docs will shadow it. I wonder whether this order of precedence is always desirable. I could imagine another potential use case that requires the opposite order of precedence: plugin-debug defines some default components, and let theme-classic or theme-bootstrap to provide better styling. Although I think the plugin-debug use case is probably less important.

Reply:
Thanks for your insights :) currently trying to figure this out by using the @theme-init alias and see if I can "forward" real useDocs impl. Not sure how to do this properly, Webpack emits a warning on conditional requires (https://github.com/webpack/webpack/issues/7713#issuecomment-467888437)

Reply:
Seems to work according to the tests but there is an annoying webpack warning, not sure how to improve this.